zebrad/src/components/mempool/crawler/tests.rs

```diff
@@ -70,4 +84,241 @@ proptest! {
             Ok::<(), TestCaseError>(())
         })?;
     }
+
+    /// Test if crawled transactions are forwarded to the [`Mempool`][mempool::Mempool] service.
+    ///
+    /// The transaction IDs sent by other peers to the crawler should be forwarded to the
+    /// [`Mempool`][mempool::Mempool] service so that they can be downloaded, verified and added to
+    /// the mempool.
+    #[test]
+    fn crawled_transactions_are_forwarded_to_downloader(
+        transaction_ids in vec(any::<UnminedTxId>(), 1..MAX_CRAWLED_TX),
+    ) {
+        let runtime = tokio::runtime::Builder::new_current_thread()
+            .enable_all()
+            .build()
+            .expect("Failed to create Tokio runtime");
+        let _guard = runtime.enter();
+
+        let transaction_id_count = transaction_ids.len();
+
+        runtime.block_on(async move {
+            let (mut peer_set, mut mempool, _sync_status, mut recent_sync_lengths) =
+                setup_crawler();
+
+            time::pause();
+
+            // Mock end of chain sync to enable the mempool crawler.
+            recent_sync_lengths.push_extend_tips_length(0);
+
+            crawler_iteration(&mut peer_set, vec![transaction_ids.clone()]).await?;
+
+            respond_to_queue_request(
+                &mut mempool,
+                transaction_ids,
+                vec![Ok(()); transaction_id_count],
+            ).await?;
+
+            mempool.expect_no_requests().await?;
+
+            Ok::<(), TestCaseError>(())
+        })?;
+    }
+
+    /// Test if errors while forwarding transaction IDs do not stop the crawler.
+    ///
+    /// The crawler should continue operating normally if some transactions fail to download or
+    /// even if the mempool service fails to enqueue the transactions to be downloaded.
+    #[test]
+    fn transaction_id_forwarding_errors_dont_stop_the_crawler(
+        service_call_error in any::<MempoolError>(),
+        transaction_ids_for_call_failure in vec(any::<UnminedTxId>(), 1..MAX_CRAWLED_TX),
+        transaction_ids_and_responses in
+            vec(any::<(UnminedTxId, Result<(), MempoolError>)>(), 1..MAX_CRAWLED_TX),
+        transaction_ids_for_return_to_normal in vec(any::<UnminedTxId>(), 1..MAX_CRAWLED_TX),
+    ) {
+        let runtime = tokio::runtime::Builder::new_current_thread()
+            .enable_all()
+            .build()
+            .expect("Failed to create Tokio runtime");
+        let _guard = runtime.enter();
+
+        runtime.block_on(async move {
+            let (mut peer_set, mut mempool, _sync_status, mut recent_sync_lengths) =
+                setup_crawler();
+
+            time::pause();
+
+            // Mock end of chain sync to enable the mempool crawler.
+            recent_sync_lengths.push_extend_tips_length(0);
```

Comment:
Same here, can use `mempool.enable()` if desired

Reply:
Same as above :/

Reply:
`SyncStatus::sync_far_from_tip` might be more readable here.

Reply:
Done in [0f6aebf](https://github.com/jvff/zebra/commit/0f6aebf540df20039cac08b2879cbaa6ca17d7a5), thanks!